add group header and frozen column (#1061)
telerik/maui-docs: controls/datagrid/aggregates/styling.md

The .NET MAUI DataGrid provides a styling functionality for its group footer, header, and column footer aggregates.

## Style the Group Header

> The [Telerik UI for .NET MAUI DataGrid]({%slug datagrid-overview%}) provides the `GroupHeaderStyle` (of type `Style` with target type `DataGridGroupHeaderAppearance`) configuration which defines the style of the `GroupHeader` and the aggregates inside the header. 

Reviewer: ```suggestion
The [Telerik UI for .NET MAUI DataGrid]({%slug datagrid-overview%}) provides the `GroupHeaderStyle` (of type `Style` with target type `DataGridGroupHeaderAppearance`) configuration, which defines the style of the `GroupHeader` and the aggregates inside the header. 
```

> To visualize the aggregates in the DataGrid group header, set the `ShowGroupHeaderAggregates` to `True`. The property is a property inside the `RadDataGrid` instance.

> ### Style the Group Header:

Reviewer: This heading duplicates the previous one.
```suggestion
```

Use the following properties to style the `GroupHeader`:

| Property  | Description |
| ------ | ------ |

> | `BackgroundColor` | Defines the color that fills the area within the header |

Reviewer: ```suggestion
| `BackgroundColor` | Defines the color that fills the area within the header. |
```

| `TextFontSize` | Defines the size for the text part of the `GroupHeader`. |
| `TextMargin` | Defines the margin for the text part of the `GroupHeader`. |

### Style the Aggregates in the Group Header

> You can style the aggregates results depending on the value set for the `GroupAggregatesAlignment` (`enum` of type `Telerik.Maui.Controls.DataGrid.DataGridGroupAggregatesAlignment`) property.

Reviewer: ```suggestion
You can style the appearance of the aggregate results based on the value of the `GroupAggregatesAlignment` (`enum` of type `Telerik.Maui.Controls.DataGrid.DataGridGroupAggregatesAlignment`) property.
```

The available properties in the `GroupHeaderStyle` are:

| Property  | Description |
| ------ | ------ |
| `AggregatesTextColor` | Defines the color for the aggregates part of the `GroupHeader` |
| `AggregatesTextFontAttributes` | Defines the font attributes for the aggregates part of the `GroupHeader` |
| `AggregatesTextFontFamily` | Defines the font family of the aggregates part of the `GroupHeader` |
| `AggregatesTextFontSize` | Defines the size of the aggregates part of the `GroupHeader` |
| `AggregatesTextMargin` | Defines the margin for the aggregates part of the `GroupHeader` |

The following example shows how to style the aggregate results in the group header when setting the `GroupHeaderStyle` in the page's resources:

<snippet id='datagrid-group-header-aggregate-styling-example'/>

To style and customize the aggregate results when setting the `GroupAggregatesAlignment` to `NextToHeader`, use the following properties:

* `GroupAggregateCellStyle` (of type `Style` with target type `DataGridGroupAggregateCellAppearance`)&mdash;Defines the style of the aggregates cell within the group header of the DataGrid. 

> * `GroupAggregateCellStyleSelector` (`IStyleSelector`)&mdash;Allows you to set a different styles of each aggregate within the group header of the DataGrid by passing the `GroupAggregateCellContext` in the selector.

Reviewer: ```suggestion
* `GroupAggregateCellStyleSelector` (`IStyleSelector`)&mdash;Allows you to set different styles for each aggregate within the group header of the DataGrid by passing the `GroupAggregateCellContext` in the selector.
```

* `GroupAggregateCellTemplate` (`DataTemplate`)&mdash;Defines the appearance of each aggregate within the group header of the DataGrid.

> > The properties are a properties inside the `RadDataGrid` instance.

Reviewer: ```suggestion
> The properties listed above are properties inside the `RadDataGrid` instance.
```

The table below describes the available properties for the `GroupAggregateCellStyle`:

| Property  | Description |
| ------ | ------ |
| `TextColor` | Defines the color of the cell text. |
| `FontAttributes` | Defines the font attributes for the text in the cell. |
| `FontFamily` | Defines the font family for the text in the cell. |
| `FontSize` | Defines the size for the text in the cell. |
| `Margin` | Defines the margin that is applied to the text of the cell. |

#### Example with the `GroupAggregateCellStyle`

**1.** Define the DataGrid in XAML:

<snippet id='datagrid-group-header-aggregates'/>

**2.** Define the `GroupAggregateCellStyle` in the page's resources:

<snippet id='datagrid-group-header-aggregate-nexttoheader-style-xaml'/>

**3.** Define the`DataModel`:

<snippet id='datagrid-aggregates-model'/>

**4.** Define the `ViewModel`:

<snippet id='datagrid-aggregates-viewmodel'/>

> >important For DataGrid `GroupHeaderAggregates Style` example refer to the [SDKBrowser Demo application]({%slug sdkbrowser-app%}) and navigate to the **DataGrid > Aggregates** category.

Reviewer: ```suggestion
>important For the DataGrid `GroupHeaderAggregates Style` example, refer to the [SDKBrowser Demo application]({%slug sdkbrowser-app%}) and navigate to the **DataGrid > Aggregates** category.
```

#### Example with the `GroupAggregateCellStyleSelector`

**1.** Define the DataGrid in XAML:

<snippet id='datagrid-group-header-aggregates'/>

**2.** Define the `StyleSelector` class:

<snippet id='datagrid-group-header-aggregate-nexttoheader-style-xaml'/>

**3.** Define the `GroupAggregateCellStyleSelector` in the page's resources:

<snippet id='datagrid-group-header-aggregate-nexttoheader-style-xaml'/>

**4.** Define the `DataModel`:

<snippet id='datagrid-aggregates-model'/>

**5.** Define the `ViewModel`:

<snippet id='datagrid-aggregates-viewmodel'/>

> >important For DataGrid `GroupHeaderAggregates StyleSelector` example refer to the [SDKBrowser Demo application]({%slug sdkbrowser-app%}) and navigate to the **DataGrid > Aggregates** category.

Reviewer: ```suggestion
>important For the DataGrid `GroupHeaderAggregates StyleSelector` example, refer to the [SDKBrowser Demo application]({%slug sdkbrowser-app%}) and navigate to the **DataGrid > Aggregates** category.
```

<snippet id='datagrid-group-aggregate-style-selector-xaml'/>

> >important For DataGrid `GroupFooter` Aggregates styling example refer to the [SDKBrowser Demo application]({%slug sdkbrowser-app%}) and navigate to the **DataGrid > Aggregates** category.

Reviewer: ```suggestion
>important For the DataGrid `GroupFooter` Aggregates styling example, refer to the [SDKBrowser Demo application]({%slug sdkbrowser-app%}) and navigate to the **DataGrid > Aggregates** category.
```

![Group Aggregate Style](../images/datagrid-grouping-aggregates.png)

> >important For DataGrid `GroupFooter` styling example refer to the [SDKBrowser Demo application]({%slug sdkbrowser-app%}) and navigate to the **DataGrid > Aggregates** category.

Reviewer: ```suggestion
>important For the DataGrid `GroupFooter` styling example, refer to the [SDKBrowser Demo application]({%slug sdkbrowser-app%}) and navigate to the **DataGrid > Aggregates** category.
```
